WeScan/Scan/CaptureSessionManager.swift

```diff
@@ -144,10 +147,67 @@ final class CaptureSessionManager: NSObject, AVCaptureVideoDataOutputSampleBuffe
             return
         }
         
-        let videoOutputImage = CIImage(cvPixelBuffer: pixelBuffer)
-        let imageSize = videoOutputImage.extent.size
+        setImageOrientation()
+        
+        let finalImage = CIImage(cvPixelBuffer: pixelBuffer)
+        let imageSize = finalImage.extent.size
         
-        guard let rectangle = RectangleDetector.rectangle(forImage: videoOutputImage) else {
+        if #available(iOS 11.0, *) {
+            VisionRectangleDetector.rectangle(forImage: finalImage) { (rectangle) in
```

Comment:
On iOS 11, nil is never called on the completion callback when no rectangle is detected. 
This means that the `noRectangleCount` is never increased. As a result, if you move your camera from an area with a rectangle to one without, the rectangle detected on the previous are stays displayed on screen.

Reply:
This is fixed in the latest commit (not pushed yet) by returning `nil` when the results array `isEmpty`.